### PR TITLE
Fix behavior of str method of Expr

### PR DESCRIPTION
**Pull request by unnonouno**
- Show quatations of str constants
- Show values of list and tuple constants
- Fix behavior of GetAttr by second arguments

### CODE REVIEW

File: chainer/utils/type_check.py [hunk 1]
```diff
@@ -220,6 +220,22 @@ def _eval_expr(v):
...
+        return str(v)
+    elif isinstance(v, list):
+        return '[{0}]'.format(', '.join(map(_repr, v)))
+    elif isinstance(v, tuple):
+        if len(v) == 0:
+            return '()'
+        elif len(v) == 1:
+            return '({0},)'.format(_repr(v[0]))
+        else:
+            return '({0})'.format(', '.join(map(_repr, v)))
```
Comment (beam2d) [inline chainer/utils/type_check.py]: How about delegating the stringization here to tuple.__str__ like `str(tuple(map(_repr, v)))`?

Comment (unnonouno) [inline chainer/utils/type_check.py]: As a tuple uses `__repr__` method in its `__str__`, this makes `('1',)` for `(1,)`.

File: tests/utils_tests/test_type_check.py [hunk 2]
```diff
@@ -60,7 +64,7 @@ def setUp(self):
 
     def test_str(self):
         self.assertEqual('x[1]', str(self.v1))
-        self.assertEqual('y[a]', str(self.v2))
+        self.assertEqual('y[\'a\']', str(self.v2))
```
Comment (beam2d) [inline tests/utils_tests/test_type_check.py]: You can use double quotes here to avoid escaping for readability: `"y['a']"`.

**Comment by beam2d** [comment]
LGTM except some comments.

File: chainer/utils/type_check.py (hunk 1 above)
Comment (beam2d) [inline chainer/utils/type_check.py]: Got it. Then above impl is necessary. Thank you for the reply.

**Comment by beam2d** [comment]
Thank you!

**Pull request merged**